api/agent/agent.go

```diff
@@ -847,24 +821,56 @@ func (c *container) CPUs() uint64                   { return c.cpus }
 
 // WriteStat publishes each metric in the specified Stats structure as a histogram metric
 func (c *container) WriteStat(ctx context.Context, stat drivers.Stat) {
-
-	// Convert each metric value from uint64 to float64
-	// and, for backward compatibility reasons, prepend each metric name with "docker_stats_fn_"
-	// (if we don't care about compatibility then we can remove that)
-	var metrics = make(map[string]float64)
 	for key, value := range stat.Metrics {
-		metrics["docker_stats_fn_"+key] = float64(value)
+		stats.Record(ctx, stats.FindMeasure("docker_stats_"+key).(*stats.Int64Measure).M(int64(value)))
```

Comment:
We probably don't need 'docker_stats' prefix and any append here... But we can tweak later... The driver could simply return an array (and perhaps prefix them with 'container_'  or 'docker_stats' already) instead of this map too... Lots of unnecessary translations we do for stats. TODO & not important...

Reply:
dashboard number 3 in our examples relies on this format, was trying to minimize name breakages in this PR for digestibility's sake - we need to define / finalize probably every stat we're exporting and decide on them at some point.